composer/utils/module_surgery.py

```diff
@@ -210,6 +215,16 @@ def replace_module_classes(
     return replaced_pairs
 
 
+def _infer_device(module: torch.nn.Module) -> Optional[torch.device]:
+    """Attempt to infer a module's device by inspecting its parameters and buffers."""
+    try:
+        p = next(itertools.chain(module.parameters(), module.buffers()))
+    except StopIteration:
+        return None
+
+    return p.device
```

Comment:
Nit: Prefer putting `return p.device` inside `try` -- even though there's only one code path, imo it's better to return while its clear `p` is in scope so its more readable.

Reply:
done!